modules/gltools/src/state-tracker/webgl-parameter-tables.ts

```diff
@@ -84,12 +100,40 @@ const enable = (gl, value, key) => (value ? gl.enable(key) : gl.disable(key));
 const hint = (gl, value, key) => gl.hint(key, value);
 const pixelStorei = (gl, value, key) => gl.pixelStorei(key, value);
 
-const drawFramebuffer = (gl, value) => {
-  const target = isWebGL2(gl) ? GL.DRAW_FRAMEBUFFER : GL.FRAMEBUFFER;
+const bindFramebuffer = (gl, value, key) => {
+  let target;
+  if (key === GL.FRAMEBUFFER_BINDING) {
+    target = isWebGL2(gl) ? GL.DRAW_FRAMEBUFFER : GL.FRAMEBUFFER;
+  } else {
+    // GL.READ_FRAMEBUFFER_BINDING
+    target = GL.READ_FRAMEBUFFER;
+  }
   return gl.bindFramebuffer(target, value);
 };
-const readFramebuffer = (gl, value) => {
-  return gl.bindFramebuffer(GL.READ_FRAMEBUFFER, value);
+
+const bindBuffer = (gl, value, key) => {
+  const target = ({
+    [GL.ARRAY_BUFFER_BINDING]: [GL.ARRAY_BUFFER],
+    [GL.COPY_READ_BUFFER_BINDING]: [GL.COPY_READ_BUFFER],
+    [GL.COPY_WRITE_BUFFER_BINDING]: [GL.COPY_WRITE_BUFFER],
+    [GL.PIXEL_PACK_BUFFER_BINDING]: [GL.PIXEL_PACK_BUFFER],
+    [GL.PIXEL_UNPACK_BUFFER_BINDING]: [GL.PIXEL_UNPACK_BUFFER],
+    [GL.TRANSFORM_FEEDBACK_BUFFER_BINDING]: [GL.TRANSFORM_FEEDBACK_BUFFER],
```

Comment:
Note that transform feedback buffers and uniform buffers bindings are more complex. There are multiple binding points for these and also buffer ranges can be specified during binding. I suspect we may want to blacklist these for now. See
- https://developer.mozilla.org/en-US/docs/Web/API/WebGL2RenderingContext/bindBufferBase
- https://developer.mozilla.org/en-US/docs/Web/API/WebGL2RenderingContext/bindBufferRange

Update: I note you have black listed these below, to avoid confusion, should we still list them here or at least add a comment saying they are black listed?

Reply:
Looks like I missed these when I removed them from the defaults.